FIRO_TSEnsembles/src/main/java/hec/SqliteDatabase.java

```diff
@@ -566,6 +569,42 @@ public void write(MetricCollectionTimeSeries[] mtsArray) throws Exception {
         }
         _connection.commit();
     }
+
+    private Boolean doesTableExist(String tableName) {
+
+        PreparedStatement p = null;
+        try{
+            String sql = "SELECT name FROM sqlite_master WHERE type='table' AND name= ?";
+            p = _connection.prepareStatement(sql);
+            p.setString(1, tableName);
+            ResultSet rs = p.executeQuery();
+            rs.next();
+            return rs.getString("name").equals(tableName);
+        } catch (Exception e) {
+            Logger.logError(e);
+        }
+        return false;
+    }
+
+    public void deleteAllEnsemblesFromDB() throws Exception{
+
+        if (doesTableExist("ensemble")) {
+            if (ps_drop_ensemble == null) {
+                String sql1 = "DELETE from ensemble";
+                ps_drop_ensemble = _connection.prepareStatement(sql1);
+                String sql2 = "DELETE from ensemble_timeseries";
+                ps_drop_ensemble_timeseries = _connection.prepareStatement(sql2);
+            }
+            if (ps_drop_ensemble_timeseries == null) {
+                String sql2 = "DELETE from ensemble_timeseries";
+                ps_drop_ensemble_timeseries = _connection.prepareStatement(sql2);
+            }
+            ps_drop_ensemble.execute();
+            ps_drop_ensemble_timeseries.execute();
+        }
```

Comment:
@ktarbet do you need to `_connection.commit();` after you execute these queries?

Reply:
@ktarbet from my testing in ResSim I found I needed commit these changes.  I tried to find a way to vacuum the database, but got stuck because it could not be performed during a transaction.